### PR TITLE
[ETFE-4269] block GB00 and XI00 users from CoD gracefully

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/controllers/actions/AuthAction.scala [hunk 1]
```diff
@@ -106,4 +109,21 @@ class AuthActionImpl @Inject()(override val authConnector: AuthConnector,
...
+                                            internalId: String,
+                                            credId: String,
+                                            sessionId: Option[SessionId],
+                                            hasMultipleEnrolments: Boolean
+                                           )(block: UserRequest[A] => Future[Result])
+                                           (implicit request: Request[A]): Future[Result] = {
+    val userType = UserType(ernFromUrl)
+
+    if (Seq(Unknown, GreatBritainWarehouse, NorthernIrelandWarehouse).contains(userType)) {
+      logger.warn(s"[checkIfUserErnCanAccessCaM] User attempted to access CaM with invalid ern: '$ernFromUrl'")
```
Comment (grahampaulcook) [inline app/controllers/actions/AuthAction.scala:123]:
```suggestion
      logger.warn(s"[checkIfUserErnCanAccessCOD] User attempted to access COD with invalid ern: '$ernFromUrl'")
```

File: app/controllers/actions/AuthAction.scala [hunk 2]
```diff
@@ -97,7 +99,8 @@ class AuthActionImpl @Inject()(override val authConnector: AuthConnector,
       case emcsEnrolments =>
         emcsEnrolments.find(_.identifiers.exists(ident => ident.key == EnrolmentKeys.ERN && ident.value == ernFromUrl)) match {
           case Some(enrolment) if enrolment.isActivated =>
-            block(UserRequest(request, ernFromUrl, internalId, credId, sessionId.get.value, emcsEnrolments.size > 1))
+            //            block(UserRequest(request, ernFromUrl, internalId, credId, sessionId.get.value, emcsEnrolments.size > 1))
```
Comment (grahampaulcook) [inline app/controllers/actions/AuthAction.scala:102]: Probably need to remove the commented line?

Comment (james-work-account) [reply app/controllers/actions/AuthAction.scala]: Addressed in https://github.com/hmrc/emcs-tfe-change-destination-frontend/pull/57/commits/f2d17b54feea4abda070e4d59e4a8b420c3e0a92

File: app/controllers/actions/AuthAction.scala (hunk 1 above)
Comment (james-work-account) [reply app/controllers/actions/AuthAction.scala]: Addressed in https://github.com/hmrc/emcs-tfe-change-destination-frontend/pull/57/commits/9bbd4de2d48fe7682f5b386720efe2dd1575e9cf

**Review by grahampaulcook — APPROVED** [review]

**Pull request merged**